packages/realm-react/src/__tests__/AppProvider.test.tsx

```diff
@@ -0,0 +1,70 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import React, { useState } from "react";
+import { renderHook } from "@testing-library/react-hooks";
+import { AppProvider, useApp } from "../AppProvider";
+import { View, Text, Button } from "react-native";
+import { act, fireEvent, render, waitFor } from "@testing-library/react-native";
+
+describe("AppProvider", () => {
+  it("returns the configured app with useApp", async () => {
+    const wrapper = ({ children }: { children: React.ReactNode }) => (
+      <AppProvider id="someId" app={{ name: "someName", version: "42" }} baseUrl="someurl">
+        {children}
+      </AppProvider>
+    );
+    const { result } = renderHook(() => useApp(), { wrapper });
+    const app = result.current;
+    expect(app.id).toBe("someId");
```

Comment:
Should we also check the app `name`, `version` and `baseUrl` (assuming there's a way to access them)?

Reply:
Oddly enough, they are not accessible

Reply:
Ah cool, not to worry